src/Contract_Item.php

```diff
@@ -482,10 +496,52 @@ public static function showForContract(Contract $contract, $withtemplate = 0)
                 foreach ($iterator as $objdata) {
                     $data[$itemtype][$objdata['id']] = $objdata;
                     $used[$itemtype][$objdata['id']] = $objdata['id'];
+                    $totalnb++;
                 }
             }
         }
 
+        $contract_users_table = Contract_User::getTable();
+        $users_table = User::getTable();
+        $entity_table = Entity::getTable();
+
+        // Add contract users
+        $user_params = [
+            'SELECT' => [
+                "$users_table.*",
+                "$contract_users_table.id AS linkid",
+                "$entity_table.id AS entity"
```

Comment:
IMHO, the `entities_id` field should not be filled for users. Indeed, for all other itemtypes, it indicates the entity where the item is visible. For users, it indicates the default entity used at connection, and this information will be useless here, and can even be confusing.

```suggestion
```